File: simplypdf/src/main/java/com/wwdablu/soumya/simplypdf/composers/models/cell/Cell.java
package com.wwdablu.soumya.simplypdf.composers.models.cell;

public abstract class Cell {

    public int width;
    public int horizontalPadding;
    public int verticalPadding;

    Cell() {
        horizontalPadding = 10;
        verticalPadding = 10;
    }

    public int getHorizontalPadding() {
        return horizontalPadding;
    }

    public void setHorizontalPadding(int horizontalPadding) {
        this.horizontalPadding = horizontalPadding;
    }

    public int getVerticalPadding() {
        return verticalPadding;
    }

    public void setVerticalPadding(int verticalPadding) {
        this.verticalPadding = verticalPadding;
    }
}

File: simplypdf/src/main/java/com/wwdablu/soumya/simplypdf/composers/models/cell/Cell.kt
package com.wwdablu.soumya.simplypdf.composers.models.cell

import com.wwdablu.soumya.simplypdf.SimplyPdfDocument

abstract class Cell internal constructor(
    var simplyPdfDocument: SimplyPdfDocument
) {

    var horizontalPadding = 10
    var verticalPadding = 10

    abstract fun getCellWidth() : Int
    abstract fun getCellHeight() : Int
    abstract fun render(xTranslate: Int) : Boolean
}